viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/tree.tsx

```diff
@@ -118,7 +131,49 @@ export class FilterTree extends React.Component<FilterTreeProps> {
 
     isCollapsed = (id: number): boolean => this.props.collapsedNodes.includes(id);
 
-    renderTreeNodes = (nodes: TreeNode[], _parent: TreeNode = defaultTreeNode, level = 0): JSX.Element | undefined => {
+    handleFilterChanged = (filter: string) : void => {
+        let filteredTree: TreeNode[] = [];
+        const matchedIds: number[] = [];
+        const rootNodes = this.getRootNodes();
+        rootNodes.forEach((node: TreeNode) => this.getMatchingIds(node, filter, matchedIds));
+        filteredTree = this.filterTree(this.props.nodes, matchedIds);
+        this.setState({filteredNodes: filteredTree});
+    }
```

Comment:
There are many issues here that should be auto-fixed when saving the file is vscode. Maybe you need to add eslint plugin to your vscode?